lms/djangoapps/teams/views.py

```diff
@@ -889,6 +888,13 @@ def post(self, request):
         except User.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
+        course_module = modulestore().get_course(team.course_id)
+        if course_module.teams_max_size and team.users.count() >= course_module.teams_max_size:
+            return Response(
```

Comment:
I tried this, it's causing a lot more failures in python tests for cases where teams_max_size is never defined. I think we need to avoid comparing against None?